### PR TITLE
[metrics] Only generate metrics if explicitly requested

### PR DESCRIPTION
**Pull request by jaylees14**
Check if the user requested metrics and generate accordingly.

I decided to not include metrics in the simple project builder test, but I've added them to the complex one (bit of variety eh!) - hence the large negative diff.

### CODE REVIEW

File: src/main/scala/temple/builder/project/ProjectBuilder.scala [hunk 1]
```diff
@@ -74,8 +74,15 @@ object ProjectBuilder {
   }
 
   private def buildMetrics(templefile: Templefile): Files = {
-    // TODO: Get this from templefile and project settings
-    val datasource: Datasource = Datasource.Prometheus("Prometheus", "http://prom:9090")
+    // Only generate metrics if explicitly given
+    val metric = templefile.lookupMetadata[Metrics].getOrElse {
+      return Map.empty
```
Comment (c-harding) [inline src/main/scala/temple/builder/project/ProjectBuilder.scala:80]: please can we explicitly comment this line to explain the early return?

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

**Review by c-harding — APPROVED** [review]
[Relevant gif that didn’t display properly](https://media.giphy.com/media/fnMUG7ybQ8hwRgicfN/giphy.gif)

**Comment by c-harding** [comment]
PR message needs fixing
